script.js

```diff
@@ -1,4 +1,147 @@
+var gameState = 0;
+var diceCombinedPlayerOneFirst = [];
+var diceCombinedPlayerOneSecond = [];
+var diceCombinedPlayerTwoFirst = [];
+var diceCombinedPlayerTwoSecond = [];
+var playerOneFinalResult = "";
+var playerTwoFinalResult = "";
+
+//START OF THE GAME//
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var output = document.querySelector("#submit-button");
+  if (gameState === 0 && input === "") {
+    gameState = "start";
+    output.innerHTML = "Roll";
+    return "Welcome to the game of Beat That!<br><br>This is a 2-player game. You will roll 2 dice, and you can choose the order of the dice to form the largest number. The player with the largest number wins!<br><br>🎲 Player 1, please roll the dice. 🎲";
+  }
+  if (input != "1" && input != "2" && input != "") {
+    return "Error, please only key '1' or '2' to choose your dice order.";
+  }
+  //PLAYER 1 STARTS GAME//
+  if (gameState === "start" && input === "") {
+    gameState = "player1";
+    output.innerHTML = "Enter";
+    return playerStartGame(
+      diceCombinedPlayerOneFirst,
+      diceCombinedPlayerOneSecond,
+      "1"
+    );
```

Comment:
this is great! would like you to remember this pattern for future use. But in this scenario, we have `diceCombinedPlayerOneFirst` and `diceCombinedPlayerOneSecond` in the global scope, which means `playerStartGame` function is able to access it directly too and you dont have to pass it through the function